c7n/resources/asg.py

```diff
@@ -988,6 +988,49 @@ def process_asg(self, asg):
             raise
 
 
+@actions.register('modify-security-groups')
+class ASGModifyGroups(ModifyGroupsAction):
+    # Need some kind of a warning - this will require the ASGs to bring up new instances
+    # in order to see the new security groups
+
+    # TODO: Implement. Lots of other gotchas
+
+    def process(self, asgs):
+        raise NotImplemented('The modify-security-groups action has not been implemented on this resource')
+        with self.executor_factory(max_workers=5) as w:
+            list(w.map(self.process_asg, asgs))
+
+    def process_asg(self, asg):
+        client = local_session(self.manager.session_factory).client('autoscaling')
+
+        # patch the resource obj for the generic actions
+        asg['Groups'] = asg['SecurityGroups']
+
+        groups = super(ASGModifyGroups, self).get_groups(asg)
+
+        # create new launch config from old
+        # May be a better way to do these describe calls - batch them together
+        config_name = asg['LaunchConfigurationName']
+        old_config = client.describe_launch_configurations(
+                LaunchConfigurationNames=[config_name]
+        )['LaunchConfigurations'][0]
+
+        # other than these two, the describe is identical to create syntax. convenient
+        new_config = old_config.pop('LaunchConfigurationARN').pop('CreatedTime').pop('LaunchConfigurationName')
+        new_config['SecurityGroups'] = groups
+        # TODO: come up with a way to generate a new launch config name
+        new_launch_config_name = ''
+        new_config['LaunchConfigurationName'] = new_launch_config_name
+
+        client.create_launch_configuration(**new_config)
+
+        # Update ASG with new config
+        # something with client.update(config_name)
+        client.update_auto_scaling_group(
```

Comment:
future suggestion, asg launch-config updates could be their own helper func. lots of other use-cases.